lumberjack.go

```diff
@@ -356,6 +379,59 @@ func (l *Logger) oldLogFiles() ([]logInfo, error) {
 	return logFiles, nil
 }
 
+// compressLog compresses the log with given filename
+// using Gzip compression
+func compressLog(filename string) error {
+
+	// If a blank filename is passed
+	// return false
+	if filename == "" {
```

Comment:
You don't really need this.  This is not a general function that will be used by third parties... it's used in exactly one place, and we know the filename won't be empty.  Plus, os.Open will return an error if you try to open an empy filename.